File: PROJETOS/20180712-useacabeca-c/cap2.5/jukebox.go
package main

import (
	"fmt"
	"strings"
)

var tracks = []string{
	"I left my  hearth in Harvard Med School",
	"Newark, Newark - a wonderful town",
	"Dancing with a Dork",
	"From here to maternity",
	"The girl from Iwo Jima",
}

func find_track(query string) {
	for i := 0; i < 5; i++ {
		if strings.Index(tracks[i], query) != -1 {
			fmt.Printf("Found: %s\n", tracks[i])
		}
	}
}

func main() {
	fmt.Printf("Pesquisar por: ")
	var query string
	fmt.Scan(&query)
	find_track(query)
}

File: PROJETOS/20180712-useacabeca-c/cap2.5/jukebox.c
#include <stdio.h>
#include <string.h>

char tracks[][80] = {
 "I left my heart in Harvard Med School",
 "Newark, Newark - a wonderful town",
 "Dancing with a Dork",
 "From here to maternity",
 "The girl from Iwo Jima",
};

void find_track(char query[])  {
    int i;
    for (i = 0; i < 5; i++) {
        if (strstr(tracks[i], query)) {
            //printf("Achado id:%i: %s\n", i, tracks[i]);
            puts(tracks[i]);
        }
    }
}

int main () {
    char query[80];
    printf("Pesquisar por?: ");
    scanf("%80s", query);
    //fgets(query, 80, stdin); // apenas com scanf funciona :/
    find_track(query);
    return 0;
}
